torchgeo/datasets/spacenet.py

```diff
@@ -191,49 +199,287 @@ def __getitem__(self, index: int) -> Dict[str, Tensor]:
+            return []
+
+        to_be_downloaded = []
+        for collection in missing_collections:
+            archive_path = os.path.join(self.root, collection + ".tar.gz")
+            if os.path.exists(archive_path):
+                print(f"Found {collection} archive")
+                if (
+                    self.checksum
+                    and check_integrity(
+                        archive_path, self.collection_md5_dict[collection]
+                    )
+                    or not self.checksum
+                ):
+                    print("Extracting...")
+                    extract_archive(archive_path)
+                else:
+                    print(f"Collection {collection} is corrupted")
+                    to_be_downloaded.append(collection)
+            else:
+                print(f"{collection} not found")
+                to_be_downloaded.append(collection)
+
+        return to_be_downloaded
+
+    def _download(self, collections: List[str], api_key: Optional[str] = None) -> None:
         """Download the dataset and extract it.
 
         Args:
+            collections: Collections to be downloaded
             api_key: a RadiantEarth MLHub API key to use for downloading the dataset
 
         Raises:
             RuntimeError: if download doesn't work correctly or checksums don't match
         """
-        if self._check_integrity():
-            print("Files already downloaded")
-            return
-
-        download_radiant_mlhub_dataset(self.dataset_id, self.root, api_key)
-        archive_path = os.path.join(self.root, self.foldername + ".tar.gz")
-        if (
-            self.checksum
-            and check_integrity(archive_path, self.md5)
-            or not self.checksum
-        ):
+        for collection in collections:
+            download_radiant_mlhub_collection(collection, self.root, api_key)
+            archive_path = os.path.join(self.root, collection + ".tar.gz")
+            if (
+                not self.checksum
+                or not check_integrity(
+                    archive_path, self.collection_md5_dict[collection]
+                )
+            ) and self.checksum:
+                raise RuntimeError(f"Collection {collection} corrupted")
+
+            print("Extracting...")
             extract_archive(archive_path)
-        else:
-            raise RuntimeError("Dataset corrupted")
+
+
+class SpaceNet1(SpaceNet):
+    """SpaceNet 1: Building Detection v1 Dataset.
+
+    `SpaceNet 1 <https://spacenet.ai/spacenet-buildings-dataset-v1/>`_
+    is a dataset of building footprints over the city of Rio de Janeiro.
+
+    * *Dataset features*:
+
+        * No. of images: 6940 (8 Band) + 6940 (RGB)
+        * No. of polygons: 382,534 building labels
+        * Area Coverage: 2544 sq km
+        * GSD: 1 m (8 band),  50 cm (rgb)
+        * Chip size: 102 x 110 (8 band), 407 x 439 (rgb)
+
+    .. note::
+       Chip size of both imagery can have 1 pixel difference
```

Comment:
We should fix this later

Reply:
We could  resize it in `__getitem__` to a fixed size before returning it. For example,

Instead of this
```
if self.transforms is not None:
    sample = self.transforms(sample)

return sample
```

we could do this


```
from torchvision import transforms as T
....

if self.image == "8band":
    size = (102, 110)
else:
    size = (407, 439)

rsz = T.Resize(size)
sample["image"] = rsz(sample["image"])
sample["mask"] = rsz(sample["mask"])

if self.transforms is not None:
    sample = self.transforms(sample)

return sample
```

Reply:
Since it's only a 1 pixel difference, I would actually just index it instead of resizing it, should be much much faster. Do you want to do that in this PR or in a follow-up PR? I'm trying to get this merged sooner rather than later since you can't work on SpaceNet 3+ until this gets merged.

Reply:
Could you elaborate on what you mean by index an image?

Reply:
Like if the images are 120x150, 121x150, and 120x151, I would just return:
```python
image[:120,:150]
```

Reply:
Ah right. Yeah that would be faster. I'll do that in a follow up PR.